spec/models/database_spec.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe "Database" do
+  let(:models_todo) {
+    ["Spree::CreditCard", "Spree::Adjustment",
+     "StripeAccount", "ColumnPreference",
+     "Spree::LineItem", "Spree::ShippingMethod",
+     "Spree::ShippingRate"].freeze
+  }
+
+  it "should have foreign keys for models with a belongs_to relationship" do
+    Rails.application.eager_load!
+    model_classes = filter_model_classes
+
+    migrations = generate_migrations(model_classes)
+
+    expect(migrations.length).to eq(0)
+  end
+
+  def filter_model_classes
+    Dir.glob(Rails.root.join('app/models/**/*.rb').to_s)
+      .map do |file|
+        relative_path = Pathname.new(file).relative_path_from(Rails.root.join('app/models')).to_s
+        subdirectory = File.dirname(relative_path)
+        base_name = File.basename(file, '.rb').camelize
+        subdirectory == "." ? base_name : "#{subdirectory.camelize}::#{base_name}"
+      end
+  end
+
+  def generate_migrations(model_classes)
+    migrations = []
+    filter = lambda { |model| models_todo.include?(model) }
+    pending_models = model_classes.select(&filter)
+    model_classes.reject!(&filter)
+
+    ActiveRecord::Base.descendants.each do |model_class|
+      next unless model_classes.include?(model_class.name)
+
+      model_class.reflect_on_all_associations(:belongs_to).each do |association|
+        migration = process_association(model_class, association)
+        migrations << migration unless migration.nil?
+      end
+    end
+
+    print_missing_foreign_key_warnings(migrations)
+
+    puts "The following models are marked as todo in #{__FILE__}:"
+    puts pending_models.join(", ")
+
+    migrations
+  end
+
+  def print_missing_foreign_key_warnings(migrations)
+    return if migrations.empty?
+
+    puts "Foreign key(s) appear to be absent from the database. " \
+         "You can add it/them using the following migration(s):"
+    puts migrations.join("\n")
+    puts "\nTo disable this warning, add the class name(s) of the model(s) to models_todo " \
+         "in #{__FILE__}"
+
+    return if ENV.fetch("OFN_WRITE_FOREIGN_KEY_MIGRATIONS", false)
```

Comment:
I probably would abstract this check to a private method, that way you could cache the ENV.fetch, and more importantly it ensures they're fetched with the same condition (eg i could accidentally delete a character from the var name and not notice).